kin-recovery/recovery/src/main/java/kin/recovery/restore/presenter/RestorePresenterImpl.java

```diff
@@ -113,20 +124,23 @@ public void closeFlow() {
 
 	@Override
 	public void previousStep() {
-		switch (currentStep) {
-			case STEP_UPLOAD:
-				getView().close();
-				break;
-			case STEP_ENTER_PASSWORD:
-				getView().navigateBack();
-				getView().closeKeyboard();
-				break;
-			case STEP_RESTORE_COMPLETED:
-				getView().navigateBack();
-				break;
-			case STEP_FINISH:
-				getView().navigateBack();
-				break;
+		RestoreView view = getView();
+		if (view != null) {
```

Comment:
was there a particular bug, or it is just for safety?

Reply:
No, Yossi Segev just wanted it to be more defensive